framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInference.java

```diff
@@ -30,12 +30,22 @@
      * Updates the parameter types of the constructor {@code constructorElt} based on the arguments
      * in {@code objectCreationNode}.
      *
+     * <p>For each parameter in constructorElt:
+     *
+     * <ul>
+     *   <li>If there is no stored annotated type for that parameter, then use the type of the
+     *       corresponding argument in the object creation call objectCreationNode.
+     *   <li>If there was a stored annotated type for that parameter, then its new type will be the
+     *       LUB between the previous type and the type of the corresponding argument in the object
+     *       creation call.
+     * </ul>
+     *
      * @param objectCreationNode the Node that invokes the constructor
      * @param constructorElt the Element of the constructor
      * @param atf the annotated type factory of a given type system, whose type hierarchy will be
      *     used to update the constructor's parameters' types
      */
-    void updateInferredConstructorParameterTypes(
+    void updateFromObjectCreation(
```

Comment:
These renames are technically a breaking refactor, but I would guess no external checkers use them though.

Reply:
Yes, that is my thought as well.